lib/auth0/callbackHandler.js

```diff
@@ -0,0 +1,83 @@
+var error = require('./error');
+
+module.exports = function (options, ctx, req, res, routingInfo) {
+    if (!ctx.query.code) {
+        return options.loginError({
+            code: 401,
+            message: 'Authentication error.',
+            callbackQuery: ctx.query
+        }, ctx, req, res, routingInfo.baseUrl);
+    }
+
+    var authParams = {
+        clientId: options.clientId(ctx, req),
+        domain: options.domain(ctx, req),
+        clientSecret: options.clientSecret(ctx, req)
+    };
+    var count = !!authParams.clientId + !!authParams.domain + !!authParams.clientSecret;
+    if (count !== 3) {
+        return error({
+            code: 400,
+            message: 'Auth0 Client ID, Client Secret, and Auth0 Domain must be specified.'
+        }, res);
+    }
+
+    return require('superagent')
+        .post('https://' + authParams.domain + '/oauth/token')
+        .type('form')
+        .send({
+            client_id: authParams.clientId,
+            client_secret: authParams.clientSecret,
+            redirect_uri: routingInfo.baseUrl + '/callback',
+            code: ctx.query.code,
+            grant_type: 'authorization_code'
+        })
+        .timeout(15000)
+        .end(function (err, ares) {
+            if (err || !ares.ok) {
+                return options.loginError({
+                    code: 502,
```

Comment:
I think in that case we get a regular redirect to /callback with an error rather than code?